src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+(() => {
+  const body = document.querySelector('body');
+
+  let left = false;
+  let right = false;
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+  });
+
+  const notification = (parentNode, classes, text) => {
+    const node = document.createElement('div');
+
+    node.classList.add(...classes);
+    node.textContent = text;
+    node.dataset.qa = 'notification';
+    parentNode.append(node);
+  };
+
+  const firstPromise = new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+      left = true;
+    });
+
+    setTimeout(() => {
+      if (!left) {
+        reject(new Error('First promise was rejected'));
+      }
+    }, 3000);
+  });
+
+  const secondPromise = new Promise((resolve) => {
+    document.addEventListener('mousedown', (e) => {
+      if (e.button === 0) {
+        left = true;
+        resolve('Second promise was resolved');
+      }
+
+      if (e.button === 2) {
+        right = true;
+
+        resolve('Second promise was resolved');
+      }
+    });
+  });
+
+  const thirdPromise = new Promise((resolve) => {
+    let isResolved = false;
+
+    document.addEventListener('mousedown', (e) => {
+      if (e.button === 0 || e.button === 2) {
+        if (left && right && !isResolved) {
+          isResolved = true;
+          resolve('Third promise was resolved');
+        }
+      }
+    });
+  });
```

Comment:
The logic for the thirdPromise is incorrect. According to the task, the promise should be resolved only when both buttons have been clicked, regardless of the order. However, in your implementation, the promise will be resolved even if only one button is clicked. You should add a check to ensure that both buttons have been clicked before resolving the promise.